Core/StitchCoreSDK/Sources/StitchCoreSDK/Auth/Internal/CoreStitchAuth.swift

```diff
@@ -295,22 +264,42 @@ open class CoreStitchAuth<TStitchUser>: StitchAuthRequestClient where TStitchUse
     }
 
     /**
-     * Attempts to refresh the current access token.
+     * Performs the request necessary to refresh an access token.
      *
-     * - important: This method must be called within a lock.
+     * - return: a new APIAccessToken representing the refreshed access token.
      */
-    internal func refreshAccessToken() throws {
-        objc_sync_enter(authOperationLock)
-        defer { objc_sync_exit(authOperationLock) }
-
-        let newAccessToken = try doRefreshAccessToken()
+    internal func doRefreshAccessToken() throws -> APIAccessToken {
+        let response = try self.doAuthenticatedRequest(
+            StitchAuthRequestBuilder()
+                .withRefreshToken()
+                .with(path: self.authRoutes.sessionRoute)
+                .with(method: .post)
+                .build()
+        )
+
+        var newAccessToken: APIAccessToken!
+        do {
+            newAccessToken = try JSONDecoder().decode(APIAccessToken.self,
+                                                      from: response.body!)
+        } catch let err {
+            throw StitchError.requestError(withError: err, withRequestErrorCode: .decodingError)
+        }
 
-        self.authInfo = self.authInfo?.refresh(withNewAccessToken: newAccessToken)
+        return newAccessToken
+    }
 
-        do {
-            try self.authInfo?.write(toStorage: &self.storage)
-        } catch {
-            throw StitchError.clientError(withClientErrorCode: .couldNotPersistAuthInfo)
+    // MARK: Internal Methods
+    /*
+     * Internal method to check the list of AuthInfo objects in the list for a user with the
+     * given userId or throws if it is not found
+     */
+    internal func getUserAndIndexOrThrow(withId userId: String) throws -> (Int, AuthInfo) {
```

Comment:
[minors] Few things:
- This method should not explicitly say that it throws. That's implied in the method signature. 
- This method doesn't need to exist at all– there's no harm in doing this in line, and I would recommend it be done as such:
```swift
guard let index = loggedInUsersAuthInfo.firstIndex { $0.userID == userId },
    let user = loggedInUsersAuthInfo[index] else {
    throw StitchError.serviceError(
            withMessage: "User with id: \(userId) not found",
            withServiceErrorCode: .userNotFound)
}
```
- If you really want to keep this method, use the above code instead.